c2corg_ui/static/js/auth.js

```diff
@@ -0,0 +1,114 @@
+goog.provide('app.AuthController');
+goog.provide('app.authDirective');
+
+goog.require('app');
+
+
+/**
+ * This directive is used to manage the sign in/up forms.
+ *
+ * @return {angular.Directive} The directive specs.
+ * @ngInject
+ */
+app.authDirective = function() {
+  return {
+    restrict: 'A',
+    scope: true,
+    controller: 'appAuthController',
+    controllerAs: 'authCtrl',
+    bindToController: true
+  };
+};
+
+
+app.module.directive('appAuth', app.authDirective);
+
+
+
+/**
+ * @param {angular.Scope} $scope Scope.
+ * @param {angular.$http} $http
+ * @param {string} apiUrl Base URL of the API.
+ * @constructor
+ * @export
+ * @ngInject
+ */
+app.AuthController = function($scope, $http, apiUrl) {
+
+  /**
+   * @type {angular.Scope}
+   * @private
+   */
+  this.scope_ = $scope;
+
+  /**
+   * @type {angular.$http}
+   * @private
+   */
+  this.http_ = $http;
+
+  /**
+   * @type {string}
+   * @private
+   */
+  this.apiUrl_ = apiUrl;
+};
+
+
+/**
+ * @export
+ */
+app.AuthController.prototype.signIn = function() {
+  this.http_.post(this.buildUrl_('login'), this.scope_['signIn'], {
+    'headers': {
+      'Content-Type': 'application/json; charset=UTF-8',
+      'Accept': 'application/json, text/plain, /'
```

Comment:
@gberaudo Is the ``Accept`` value correct? (what about the trailing ``/``?). I have copied it from one of your jabber messages.